subsys/bluetooth/host/l2cap.c

```diff
@@ -1167,27 +1175,34 @@ static void le_ecred_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 	}
 
 response:
-	buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
-				      sizeof(*rsp) + (sizeof(scid) * i));
+	if (result != BT_L2CAP_LE_SUCCESS) {
+		buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
+				      sizeof(*rsp) + (sizeof(scid)));
+		rsp = net_buf_add(buf, sizeof(*rsp));
 
-	rsp = net_buf_add(buf, sizeof(*rsp));
-	(void)memset(rsp, 0, sizeof(*rsp));
+		(void)memset(rsp, 0, sizeof(*rsp));
+		memset(dcid, 0, sizeof(scid));
```

Comment:
Read the code comments above, the default is handling when some connections are refused so if you mean dcid to 0 any channel that did succeed would be set to 0.